src/fluree/db/query/fql/parse.cljc

```diff
@@ -0,0 +1,428 @@
+(ns fluree.db.query.fql.parse
+  (:require [fluree.db.query.exec :as exec]
+            [fluree.db.query.exec.where :as where]
+            [fluree.db.query.exec.select :as select]
+            [fluree.db.query.parse.aggregate :refer [parse-aggregate]]
+            [fluree.db.query.json-ld.select :refer [parse-subselection]]
+            [fluree.db.query.subject-crawl.legacy :refer [basic-to-analytical-transpiler]]
+            [fluree.db.query.fql.syntax :as syntax]
+            [clojure.string :as str]
+            [clojure.set :as set]
+            [clojure.walk :refer [postwalk]]
+            [fluree.json-ld :as json-ld]
+            [fluree.db.util.core :as util :refer [try* catch*]]
+            [fluree.db.query.analytical-filter :as filter]
+            [fluree.db.util.log :as log :include-macros true]
+            [fluree.db.dbproto :as dbproto]
+            [fluree.db.constants :as const]))
+
+#?(:clj (set! *warn-on-reflection* true))
+
+(defn basic-query?
+  [q]
+  (contains? q :from))
```

Comment:
So, looking at this and at the `syntax` ns, I'm still not sure what makes a basic query "basic." Is it just the use of the `from` key? Does that imply that there are no fancy aggregators, filters, etc, or something?

This is just a bit of internal terminology I've always been unclear about.

Reply:
I am also a little unclear, but, from what I could gather from reading the docs, looking at the code, and looking at examples, is that the basic query has a `:from` key while the analytical queries do not.

I think we should eventually treat them as two separate query languages with separate validation logic, because they are separate languages.

Reply:
We made the decision previously to deprecate basic queries. Basic queries (as it became unfortunately known as) mimicked a simple SQL query so were approachable as a first query for someone new to Fluree. It also made doing a query for a single subject super easy - something we said we could alternatively still support in a different way as per some ticket I can't find right now (but know it exists as I remember commenting on it).

Reply:
And, we want to also support the SPARQL `FROM` and `FROM NAMED` which uses `from` in a different way than basic queries had used them. https://www.w3.org/TR/rdf-sparql-query/#specifyingDataset